plutus-ir/src/Language/PlutusIR/Compiler/Datatype.hs

```diff
@@ -223,25 +225,26 @@ mkScottTy d@(Datatype _ _ _ _ constrs) = do
         PLC.mkIterTyFun p caseTys (PLC.TyVar p resultType)
 
 -- | Make the "pattern functor" of a 'Datatype'. This is just the normal type, but with the
--- type variable for the type itself free. In the case of non-recursive datatypes this is just the
--- datatype.
--- @mkDatatypePatternFunctor List = \(a :: *) -> forall (r :: *) . r -> (a -> List a -> r) -> r@
+-- type variable for the type itself free and its type variables free.
+-- @mkDatatypePatternFunctor List = forall (r :: *) . r -> (a -> List a -> r) -> r@
```

Comment:
@michaelpj, I cherry-picked the first commit in your `imp/pir-ifix` branch and got this. It seems `-`s and `+`s should be the other way around?

Reply:
Ah, great, thanks.